src/TypeError.hs

```diff
@@ -1,65 +1,23 @@
-module TypeError where
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```

Comment:
What's the problem here?

Reply:
By default GHC will complain if you declare a typeclass instance for a type in a different module than the one in which the type itself is declared.

Personally, I think it's a good safeguard to have, but at the same time it's not a *huge* deal to break this rule--it does open up the possibility of accidentally declaring overlapping instances etc. 

Most of the time it's turned off to permit implementation swapping a la SML functors.

Reply:
As I understand it this option is only active for this module, is this correct?

Reply:
I believe so

Reply:
Rectified in `64630d75`!